src/components/ImageView/ImageView.js

```diff
@@ -613,55 +608,10 @@ export default observer(
     updateImageTransform = () => {
       const { item } = this.props;
 
-      let ratio = 1;
-
-      const imgStyle = {
-        width: item.width,
-        transformOrigin: "left top",
-        transform: 'none',
-        filter: `brightness(${item.brightnessGrade}%) contrast(${item.contrastGrade}%)`,
-      };
-
-      const imgTransform = [];
-
-      if (item.zoomScale !== 1) {
-        const { zoomingPositionX, zoomingPositionY } = item;
-
-        imgTransform.push("translate(" + zoomingPositionX + "px," + zoomingPositionY + "px)");
-        imgTransform.push("scale(" + item.resize + ", " + item.resize + ")");
-      }
-
-      if (item.rotation) {
-        const translate = {
-          90: `0, -100%`,
-          180: `-100%, -100%`,
-          270: `-100%, 0`,
-        };
-
-        // there is a top left origin already set for zoom; so translate+rotate
-        imgTransform.push(`rotate(${item.rotation}deg)`);
-        imgTransform.push(`translate(${translate[item.rotation] || "0, 0"})`);
-
-        if ([90, 270].includes(item.rotation)) {
-          // we can not rotate img itself, so we change container's size via css margin hack, ...
-          ratio = item.naturalWidth / item.naturalHeight;
-          // ... prepare image size for transform rotation and use position: absolute
-          imgStyle.width = `${ratio * 100}%`;
-        }
-      }
-
-      if (imgTransform?.length > 0) {
-        imgStyle.transform = imgTransform.join(" ");
-      }
-
       if (this.imageRef.current) {
-        Object.assign(this.imageRef.current.style, imgStyle);
+        Object.assign(this.imageRef.current.style, item.imageTransform);
```

Comment:
we update styles in two places (almost) simultaneously — here and in `style` prop during render, so check do we really need to do this twice. this change will happen before render, and that one — after.